### DISCUSSION TITLE
S.M.A.R.T. status sensor

### DISCUSSION THREAD

**Discussion by Gugulator** [dougiteixeira/proxmoxve#176] [Feature request] [answered]
first i'd like to say thank you for all improvements you've made! it is great work!
my question is: is it possible to create a S.M.A.R.T. and/or Wearout status sensor for chosen drives?

**Comment by dougiteixeira** [comment]
Are you referring to the disks? If so, there is already a Health binary sensor that represents this.

**Reply by Gugulator** [reply]
May be I need to install something in Proxmox?
Because I don't have Health sensor for node, only for one of VMs.
And even if binary Health sensor for node will be existing, what disk information it shows if I have more than 1 disk?

**Reply by dougiteixeira** [reply, marked as answer]
Are you using the latest version of the integration? Download the diagnostic file and send it here for analysis...

The sensors that are created for each disk are shown in the image below:

![image](https://github.com/dougiteixeira/proxmoxve/assets/31328123/036e2060-cd71-486b-b2b7-33835baae10b)

I have the change merged, waiting for your return to release a new version that includes some more information from the disk as shown in the image below:

![image](https://github.com/dougiteixeira/proxmoxve/assets/31328123/f6174806-0ba8-4f60-ada7-cf5f29a1f629)

If after downloading and sending your diagnostic file here with your current version, you want to update to the main branch version (just go to redownload in HACS, select main in the version and click install) to test and give me feedback before I release the stable version it helps me a lot.

**Reply by Gugulator** [reply]
Sorry, my fault. Only now i realized that there is a new update 3.10.
I was using version 3.0
Now it have sensors per each drive.
You beat me to it :-)
I want to suggest new feature, but you released it in last update :-)

**Reply by dougiteixeira** [reply]
I released version 3.2.0, if you can test it.

**Reply by Gugulator** [reply]
Nice! Now it shows temperature for both disks! Thanks a lot! 👍🏻

**Comment by scottcopus** [comment]
@dougiteixeira Proxmox itself shows the disk SMART wearout percentage on a node's Disk area, usually as the last column. Can this be exposed as a sensor value?

BTW, great add-on!

**Reply by dougiteixeira** [reply]
https://github.com/dougiteixeira/proxmoxve/discussions/272
